source/common/http/header_utility.cc

```diff
@@ -201,11 +203,15 @@ bool HeaderUtility::headerValueIsValid(const absl::string_view header_value) {
                                                              http2::adapter::ObsTextOption::kAllow);
 }
 
-bool HeaderUtility::headerNameIsValid(const absl::string_view header_key) {
+bool HeaderUtility::headerNameIsValid(absl::string_view header_key) {
   if (!header_key.empty() && header_key[0] == ':') {
+#ifdef ENVOY_NGHTTP2
     // For HTTP/2 pseudo header, use the HTTP/2 semantics for checking validity
     return nghttp2_check_header_name(reinterpret_cast<const uint8_t*>(header_key.data()),
                                      header_key.size()) != 0;
+#else
+    header_key.remove_prefix(1);
```

Comment:
@birenroy  @yanavlasov AFYK is this correct for header validation (remove the : pass normal checks)?  If so I'll land a runtime guarded feature for moving off the nghttp2 checks.